CMakeLists.txt

```diff
@@ -8,6 +8,15 @@ message(STATUS "Building with: ${CMAKE_CXX_COMPILER_ID}")
 set(CMAKE_CXX_STANDARD 17)
 set(CMAKE_CXX_STANDARD_REQUIRED True)
 
+if (CMAKE_BUILD_TYPE STREQUAL "Debug")
+	set(CMAKE_CXX_FLAGS "-g")
+	set(CMAKE_C_FLAGS "-g")
+elseif(CMAKE_BUILD_TYPE STREQUAL "Release")
+	set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Ofast -march=native -mtune=native -ftree-loop-distribution -floop-interchange -freorder-blocks-and-partition ")
+endif()
+
+
+
```

Comment:
I see you have edited the CMakeLists.txt. We actually already add debug symbols, if you look at the lines below `if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")`
GCC specific Debug and Release flags can be added below the `else()`
By default Release builds already have optimization flag -O3, we already define -ffast-math so -Ofast is redundant. I'm not sure about the other optimization flags but I believe they're included in -O3?